include/exadg/operators/operator_base.cpp

```diff
@@ -2153,6 +2157,175 @@ OperatorBase<dim, Number, n_components>::evaluate_face_integrals() const
+                                                         Number>(tmp_matrix,
+                                                                 dsp,
+                                                                 dof_indices_all_cells,
+                                                                 overlapped_cell_matrices);
+
+    // factorize and store cell matrices
+    matrices.resize(n_cells, LAPACKMatrix(dofs_per_cell));
+    for(unsigned int cell_batch_index = 0; cell_batch_index < matrix_free->n_cell_batches();
+        ++cell_batch_index)
+    {
+      unsigned int const n_filled_lanes =
+        matrix_free->n_active_entries_per_cell_batch(cell_batch_index);
+
+      for(unsigned int v = 0; v < n_filled_lanes; ++v)
+      {
+        // get overlapped cell matrix
+        auto const & overlapped_cell_matrix =
+          overlapped_cell_matrices[cell_batch_index * vectorization_length + v];
+
+        // store the cell matrix and renumber lexicographic
+        auto & lapack_matrix = matrices[cell_batch_index * vectorization_length + v];
+
+        auto const & lex = matrix_free->get_shape_info().lexicographic_numbering;
+        for(unsigned int i = 0; i < dofs_per_cell; i++)
+          for(unsigned int j = 0; j < dofs_per_cell; j++)
+            lapack_matrix.set(i, j, overlapped_cell_matrix[lex[i]][lex[j]]);
+
+        // factorize the cell matrix
+        lapack_matrix.compute_lu_factorization();
+      }
+    }
+  }
+}
+
+template<int dim, typename Number, int n_components>
+void
+OperatorBase<dim, Number, n_components>::apply_inverse_additive_schwarz_matrices(
+  VectorType &       dst,
+  VectorType const & src) const
+{
+  if(is_dg)
+    apply_inverse_block_diagonal(dst, src);
+  else
+  {
+    src.update_ghost_values();
+
+    matrix_free->template cell_loop<VectorType, VectorType>(
+      [&](auto const & matrix_free, auto & dst, auto const & src, auto const & cell_range) {
+        auto const dofs_per_cell = matrix_free.get_dofs_per_cell(this->data.dof_index);
+
+        IntegratorCell integrator =
+          IntegratorCell(matrix_free, this->data.dof_index, this->data.quad_index);
+
+        dealii::Vector<Number>                                 local_vector(dofs_per_cell);
+        dealii::AlignedVector<dealii::VectorizedArray<Number>> local_weights_vector(dofs_per_cell);
+        for(unsigned int cell = cell_range.first; cell < cell_range.second; ++cell)
+        {
+          this->reinit_cell(integrator, cell);
+
+          integrator.read_dof_values(weights);
+          for(unsigned int i = 0; i < dofs_per_cell; ++i)
+            local_weights_vector[i] = integrator.begin_dof_values()[i];
+
+          integrator.read_dof_values(src);
+
+          for(unsigned int v = 0; v < vectorization_length; ++v)
+          {
+            for(unsigned int i = 0; i < dofs_per_cell; ++i)
+              local_vector[i] = integrator.begin_dof_values()[i][v] * local_weights_vector[i][v];
```

Comment:
I think here I would again refer to the symmetric weighting (if I understand the implementation correctly ...).

Reply:
New comments 77374e24c2bc712fec7e8cd17f214cd7a49330a2